lib/mailtrap/client.rb

```diff
@@ -73,13 +108,29 @@ def request_url
       "/api/send#{sandbox ? "/#{inbox_id}" : ""}"
     end
 
-    def http_client
-      @http_client ||= Net::HTTP.new(api_host, api_port).tap { |client| client.use_ssl = true }
+    def perform_request(method, uri, body = nil)
+      http_client = Net::HTTP.new(uri.host, @api_port)
+      http_client.use_ssl = true
+      request = setup_request(method, uri.path, body)
+      response = http_client.request(request)
+      handle_response(response)
     end
 
-    def post_request(path, body)
-      request = Net::HTTP::Post.new(path)
-      request.body = body
+    def setup_request(method, path, body = nil)
+      request = case method
+                when :get
+                  Net::HTTP::Get.new(path)
+                when :post
+                  Net::HTTP::Post.new(path)
+                when :patch
+                  Net::HTTP::Patch.new(path)
+                when :delete
+                  Net::HTTP::Delete.new(path)
+                else
+                  raise ArgumentError, "Unsupported HTTP method: #{method}"
+                end
+
+      request.body = body.to_json if body
```

Comment:
Do we expect to set body for get?

Reply:
No, but we might not have body for `delete`, and sometimes event for `post` too, so don't see much sense to have different request setup methods for `get`, if tou mean that,